axonserver/src/main/java/io/axoniq/axonserver/admin/eventprocessor/requestprocessor/InstructionInformation.java

```diff
@@ -95,18 +99,34 @@ public void on(InstructionResult result) {
         if (!waitingForClients.remove(clientId)) {
             return;
         }
-        if (!result.success()) {
+        if (FAILURE.equals(result.result())) {
             ErrorCode errorCode = ErrorCode.find(result.errorCode());
             completionHandler.error(new MessagingPlatformException(errorCode, result.errorMessage()));
             completionListener.get().run();
             return;
         }
+
+        resultRef.updateAndGet(old -> old.and(result.result()));
         if (waitingForClients.isEmpty()) {
-            completionHandler.success();
+            completionHandler.success(map(resultRef.get()));
             completionListener.get().run();
         }
     }
 
+    private Result map(io.axoniq.axonserver.admin.Result result) {
+        return new Result() {
+            @Override
+            public boolean isSuccess() {
+                return SUCCESS.equals(result);
+            }
+
+            @Override
+            public boolean isAccepted() {
+                return ACK.equals(result);
```

Comment:
I am wondering if it is correct to return FALSE when it is successfully executed (that means it thas also been accepted)

Reply:
I guess this can be clarified inside the Javadoc.

Reply:
Changed isAccepted to return TRUE if the result is SUCCESS.